src/enclave-agent/src/main.rs

```diff
@@ -0,0 +1,159 @@
+use std::env;
+use std::path::Path;
+use std::sync::Arc;
+
+use clap::{crate_authors, crate_version, App, Arg};
+use protocols::{image, image_ttrpc};
+use ttrpc::asynchronous::Server;
+use ttrpc::{self, error::get_rpc_status as ttrpc_error};
+
+use async_trait::async_trait;
+use tokio::signal::unix::{signal, SignalKind};
+use tokio::sync::Mutex;
+
+use anyhow::{anyhow, Result};
+use image_rs::config::ImageConfig;
+use image_rs::image::ImageClient;
+use image_rs::snapshots;
+
+const CONTAINER_BASE: &str = "/run/enclave-cc-containers";
+const CC_IMAGE_WORK_DIR: &str = "/run/image/";
+
+// TODO: will replace with unix socket
+const SOCK_ADDR: &str = "tcp://0.0.0.0:7788";
+
+struct ImageService {
+    image_client: Arc<Mutex<ImageClient>>,
+}
+
+// // This code was taken from https://github.com/kata-containers/kata-containers/blob/CCv0/src/agent/src/rpc.rs#L143 for quote
+// A container ID must match this regex:
+//     ^[a-zA-Z0-9][a-zA-Z0-9_.-]+$
+fn verify_cid(id: &str) -> Result<()> {
+    let mut chars = id.chars();
+
+    let valid = match chars.next() {
+        Some(first)
+            if first.is_alphanumeric()
+                && id.len() > 1
+                && chars.all(|c| c.is_alphanumeric() || ['.', '-', '_'].contains(&c)) =>
+        {
+            true
+        }
+        _ => false,
+    };
+
+    match valid {
+        true => Ok(()),
+        false => Err(anyhow!("invalid container ID: {:?}", id)),
+    }
+}
+
+impl ImageService {
+    fn new() -> Self {
+        env::set_var("CC_IMAGE_WORK_DIR", CC_IMAGE_WORK_DIR);
```

Comment:
- What is this variable used by? Please document.
- Why is it set in `new()`? Why not just set it in a `setup()` function/method once?

Reply:
> * What is this variable used by? Please document.
> * Why is it set in `new()`? Why not just set it in a `setup()` function/method once?

It refers to the implementation of kata https://github.com/kata-containers/kata-containers/blob/CCv0/src/agent/src/image_rpc.rs#L52

Reply:
Since that's in a different project, I think it would be helpful to add that URL as a comment in the code.